### ISSUE TITLE
macOS CMake CI builds can't find `libSDL2_test`

### ISSUE THREAD

**Issue by smcv** [libsdl-org/SDL_image#260]
Even after #258, we're only building or running `testimage` on Linux+Autotools, Linux+CMake and macOS+Autotools, because the macOS+CMake build fails to find `SDL_test`:

```
FAILED: test/testimage 
: && /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -O3 -DNDEBUG -isysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk -mmacosx-version-min=11.6 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  test/CMakeFiles/testimage.dir/main.c.o -o test/testimage  -Wl,-rpath,/Users/runner/work/SDL_image/SDL_image/build  libSDL2_image-2.0.501.0.0.dylib  /usr/local/Cellar/sdl2/2.0.22/lib/libSDL2main.a  -lSDL2_test  /usr/local/Cellar/sdl2/2.0.22/lib/libSDL2.dylib && :
ld: library not found for -lSDL2_test
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

@madebr speculates that something like this might work:

```cmake
find_library(SDL2_TEST_LIBRARY
     NAMES SDL2_test
)
if(SDL2_TEST_LIBRARY AND NOT TARGET SDL2::SDL2test)
    add_library(SDL2::SDL2test UNKNOWN IMPORTED)
    set_target_properties(SDL2::SDL2test PROPERTIES
        IMPORTED_LOCATION "${SDL2_TEST_LIBRARY}"
    )
endif()
```

but neither of us are Mac users, so this might be wrong.

Windows is out of scope here: the MSYS mingw-w64-*-SDL packages that we're using don't even *have* `SDL_test` (it's not just that we can't find it, [it isn't even in the package](https://packages.msys2.org/package/mingw-w64-x86_64-SDL2)).

**Comment by madebr** [comment]
Fixed in https://github.com/libsdl-org/SDL_image/commit/22c4127100e81d58a913085213adbe1957f7c8a1
